### ISSUE TITLE
lint: return after fatal/panic

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#444]
Detects return statements that follow panic/fatal logging.

**Before:**
```go
log.Fatal("exits funciton")
return
```

**After:**
```go
log.Fatal("exits function")
```

**Comment by Quasilyte** [comment]
Lets keep this one for new contributors.

**Comment by breml** [comment]
My PR provides a very basic initial implementation for this. I asked my self, to what extend, this should be generalized to also cover 3rd party libraries like [logrus](https://godoc.org/github.com/sirupsen/logrus#Panic) or instances of [`log.Logger`](https://golang.org/pkg/log/#Logger). Would it be safe to warn for all occurances of `Fatal`, `Fatalf`, etc. instead of `log.Fatal`, `log.Fatalf`, etc.?

**Comment by breml** [comment]
An other thought is, that every statement after a `log.Fatal` is dead code (not only `returns`), because no code is executed after the `panic` in `log.Fatal`. Therefore we could extend this checker to warn about every statement after a `log.Fatal`.

**Comment by cristaloleg** [comment]
A 'dead' code after a `panic` is handled by `golint` afair. But we can add it too.

**Comment by Quasilyte** [comment]
@breml thanks for submitting this!

Well, we may enable adding more function names to the known list for the user.
Basic implementation could only handle most frequently used functions that do non-local return.

**Comment by breml** [comment]
I am not talking about 'dead' code after a direct call to `panic` but code after `log.Fatal` like this:

```
func main() {
	test := true
	if test {
		log.Fatal("end")
		fmt.Println()
	}
}
```

Here `golint` does not complain about the `fmt.Println()`, even though it will never be executed.

**Comment by breml** [comment]
@Quasilyte please wait with merging, I will make it more general in regards to "dead" code, which will make the logic even simpler.

**Comment by Quasilyte** [comment]
@breml, no problems, take your time.

**Comment by breml** [comment]
@Quasilyte OK, updated the PR such that all code following one of the *exiting functions* in the `log` package of stdlib is treated as *deadcode*. Adding further packages as well as instances like `log.Logger` will have to follow in a separate PR.
